tests/test_utils/test_generation_utils.py

```diff
@@ -3,3 +3,117 @@
 Distributed under the MIT License
 (See an accompanying file LICENSE or a copy at https://opensource.org/licenses/MIT)
 """
+
+from sc_client import client
+from sc_client.constants import sc_types
+from sc_client.models import ScAddr, ScTemplate
+
+from sc_kpm import ScKeynodes
+from sc_kpm.common import CommonIdentifiers
+from sc_kpm.utils.common_utils import create_node
+from sc_kpm.utils.generation_utils import (
+    create_oriented_set,
+    create_set,
+    create_structure,
+    wrap_in_oriented_set,
+    wrap_in_set,
+)
+from tests.common_tests import BaseTestCase
+
+
+class TestGenerationUtils(BaseTestCase):
+    def test_wrap_in_oriented_set(self):
+        set_node = create_node(sc_types.NODE_CONST)
+        start_element = create_node(sc_types.NODE_CONST)
+        other_element = create_node(sc_types.NODE_CONST)
+        wrap_in_oriented_set(set_node, start_element, other_element)
+        search_results = client.template_search(_create_oset_template(set_node, start_element, other_element))
+        assert search_results
+
+    def test_create_oriented_set(self):
+        start_element = create_node(sc_types.NODE_CONST)
+        other_element = create_node(sc_types.NODE_CONST)
+        set_node = create_oriented_set(start_element, other_element)
+        search_results = client.template_search(_create_oset_template(set_node, start_element, other_element))
+        assert search_results
+
+    def test_wrap_in_set(self):
+        set_node = create_node(sc_types.NODE_CONST)
+        element1 = create_node(sc_types.NODE_CONST)
+        element2 = create_node(sc_types.NODE_CONST)
+        wrap_in_set(set_node, element1, element2)
+        search_results = client.template_search(_create_set_template(set_node, element1, element2))
+        assert search_results
+
+    def test_create_set(self):
+        element1 = create_node(sc_types.NODE_CONST)
+        element2 = create_node(sc_types.NODE_CONST)
+        set_node = create_set(sc_types.NODE_CONST, element1, element2)
+        search_results = client.template_search(_create_set_template(set_node, element1, element2))
+        assert search_results
+
+    def test_create_structure(self):
+        element1 = create_node(sc_types.NODE_CONST)
+        element2 = create_node(sc_types.NODE_CONST)
+        set_node = create_structure(element1, element2)
+        search_results = client.template_search(_create_set_template(set_node, element1, element2))
+        assert search_results
+        assert search_results[0].addrs[0].value == set_node.value
+
+
+def _create_oset_template(set_node: ScAddr, start_element: ScAddr, other_element: ScAddr) -> ScTemplate:
```

Comment:
It's not clear, this func create only template as a code, or function generate construction in memory. What you think about:
```suggestion
def _oriented_set_template(set_node: ScAddr, start_element: ScAddr, other_element: ScAddr) -> ScTemplate:
```

Reply:
_oset_ name not correlate with _oriented_set_ for me, so maybe you can use more clarifying naming

Reply:
I'll change to `_create_oriented_set_template`